### DISCUSSION TITLE
Send `⌘` key to Vim

### DISCUSSION THREAD

**Discussion by kuntau** [wezterm/wezterm#1480] [Q&A] [answered]
How do I send <kbd>⌘</kbd> key to Vim running in wezterm.

For example <kbd>⌘</kbd>+<kbd>s</kbd> to save

**Comment by wez** [comment, marked as answer]
Only <kbd>Ctrl</kbd>, <kbd>Alt</kbd> and <kbd>Shift</kbd> modifiers have a defined encoding for the terminal.
If you wanted to, you could set up a key assignment (using https://wezfurlong.org/wezterm/config/lua/keyassignment/SendKey.html) that makes CMD-S send Ctrl-S and then you could set up vim to recognize that and trigger a save.

**Reply by kuntau** [reply]
Thank you for the answer @wez. I'm wondering, we live in 2022 why the terminal still act like it's in 80s. Can't we shake things up and make proper modern terminal. AFAIK, we also can't send combination of `Ctrl + Shift + Alt` etc to whatever we're running on.

**Reply by wez** [reply]
The issue is in getting everything to agree on what those key presses are encoded as.

There are a couple of attempts at defining keyboard encodings:
* https://sw.kovidgoyal.net/kitty/keyboard-protocol/
* http://www.leonerd.org.uk/hacks/fixterms/

but they need the applications that you run inside the terminal to detect and/or request that those encodings be used.
There isn't widespread support for either of those schemes at this time, because most terminal programs make use of
*curses* based text interfaces that have no concept of additional modifiers.
